minio/xml.py

```diff
@@ -80,20 +92,43 @@ def findtext(element, name, strict=False):
     return element.text or ""
 
 
-def unmarshal(cls, xmlstring):
+K = TypeVar("K")
+
+
+class FromXmlType(Protocol):
+    """typing stub for class with `fromxml` method"""
+
+    @classmethod
+    def fromxml(cls: Type[K], element: ET.Element) -> K:
+        """Create python object with values from XML element."""
+
+
+T = TypeVar("T", bound=FromXmlType)
+
+
+def unmarshal(cls: Type[T], xmlstring: str) -> T:
     """Unmarshal given XML string to an object of passed class."""
     return cls.fromxml(ET.fromstring(xmlstring))
 
 
-def getbytes(element):
+def getbytes(element: ET.Element) -> bytes:
     """Convert ElementTree.Element to bytes."""
-    data = io.BytesIO()
-    ET.ElementTree(element).write(
-        data, encoding=None, xml_declaration=False,
-    )
-    return data.getvalue()
+    with io.BytesIO() as data:
```

Comment:
Why do we need to use `with` here?

Reply:
close file and discard the buffer.

Reply:
https://docs.python.org/3/library/io.html#io.BytesIO

Reply:
This is not necessary, gc will handle them if we don't use `with`

Reply:
Yep. For this function, it is unnecessary.